### ISSUE TITLE
ctrl + z is unstable

### ISSUE THREAD

**Issue by valentintraen** [dbeaver/cloudbeaver#3815]
### Description

Using ctrl+z seems unstable and random inside a sql editor. Sometimes it works, sometimes (often) it doesn't.

### Steps to reproduce

Write text in an SQL editor and try ctrl + z several times

### Expected/Desired Behavior

Ctrl + z should undo the last change every time

### CloudBeaver Version

25.2.2

**Comment by dariamarutkina** [comment]
Hello, @valentintraen !
Could you please share more information?
Which OS and browser are you using, and have you noticed this issue in any previous versions of the application, or is it the first time it happens?

**Comment by valentintraen** [comment]
Hi !
Tried with : 
- Windows 11 + Chrome 141.0.7390.55
- Ubuntu 24.04 + Chrome  141.0.7390.55 and Ubuntu 24.04 + FIrefox 143.0.4

I can't tell if it's due to the latest version or not.

**Comment by dariamarutkina** [comment]
Thank you very much for your reply! 💙
We’ve re-published the images on [Docker Hub](https://hub.docker.com/r/dbeaver/cloudbeaver/tags) (version 25.2.2).
Could you please update and check if the issue is resolved?

**Comment by valentintraen** [comment]
Unfortunately, this fixed the ctrl + enter problem but not the ctrl + z problem.

**Comment by dariamarutkina** [comment]
Sorry for the inconvenience!
We're trying to reproduce the issue on our side, but we might be missing some steps. Would it be possible for you to record a short video showing how the problem happens and share it with us? That would really help us investigate 🦫

**Comment by valentintraen** [comment]
And there you have it!
https://github.com/user-attachments/assets/39fd3d2a-59d8-4692-955c-176e138669a2

**Comment by dariamarutkina** [comment]
Thank you very much for your reply! 🦫
We have one more question, could you please let us know which keyboard layout you’re using?

**Comment by valentintraen** [comment]
Hi,
It's AZERTY, I'm french

Le ven. 17 oct. 2025, 11:37, Daria Marutkina ***@***.***> a
écrit :

**Comment by dariamarutkina** [comment]
Thank you very much for your responses! 💙
It seems that this issue is related to the specifics of the AZERTY keyboard layout. We’ll try to fix it in the future.🦫

**Comment by valentintraen** [comment]
Hi,
Any news ? :)

**Comment by Hiacine** [comment]
Hi

I have also an azerty keyboard.
And the ctrl + Z is not working correctly

Is there really a diffréence between azerty and qwerty users ???

**Comment by dariamarutkina** [comment]
Hello, @Hiacine!
Thank you for the details. Yes, there can be differences between AZERTY and QWERTY, which may affect shortcuts like Ctrl + Z.
We’re looking into this. Thank you for your patience  🦫

**Comment by dariamarutkina** [comment]
1.  On the QWERTY layout, the Ctrl + Z shortcut works inconsistently 🚫

https://github.com/user-attachments/assets/b468d404-1a1b-4264-94d6-58823e794e17

2.  On the AZERTY layout, the Ctrl + Z shortcut works inconsistently 🚫

https://github.com/user-attachments/assets/a3997eb3-2d72-47c1-9706-6d293b814659

3. Many shortcuts do not work: 🚫

- Escape does not work in Data Editor
- Alt + R does not work in Data Editor
- Shift + Alt + R does not work in Data Editor
- Ctrl + . does not work in Data Editor
- Ctrl + Enter does not work in SQL Editor
- Ctrl + \ does not work in SQL Editor
- Shift + Ctrl + E does not work in SQL Editor
- Ctrl + S does not work in SQL Editor
- Shift + Ctrl + / does not work in Navigation Tree

**Comment by sergeyteleshev** [comment]
the lib is fixed so we have a green light to fix this issue:
https://github.com/JohannesKlauss/react-hotkeys-hook/issues/1318

**Comment by sergeyteleshev** [comment]
I have written unit tests for our solution to the lib so the author can accept our PR
We need to wait author to accept it, then update the lib and apply useKey changes I've done in PRs attached to this ticket
https://github.com/JohannesKlauss/react-hotkeys-hook/pull/1317#issuecomment-4769522737

**Comment by sergeyteleshev** [comment]
The author of the lib has merged our PR so we have a green light to apply our changes. Which I did and everything seems fine

**Comment by sergeyteleshev** [comment]
**For QA:**
required retest of all shortcuts

I think QWERTY & AZERTY keyboards are enough to test it

**Comment by dariamarutkina** [comment]
4. Shortcuts do not work:  (QWERTY) 🚫
- Add a new Row (ALT + R)
- Duplicate row (SHIFT + ALT + R)
- Execute SQL statement (CTRL + ↵)
- Execute SQL statement in the new Tab CTRL + \
- Show Execution plan SHIFT + CTRL + E
- Format script SHIFT + CTRL + F
- Save CTRL + S
- Redo CTRL + Y

**Comment by dariamarutkina** [comment]
Checked CE\EE ✅
Verified all shortcuts on both QWERTY and AZERTY keyboard layouts on Windows and macOS

The fix will be included in the upcoming 26.1.3 release, planned for July 20 🦫
